.github/workflows/ci.yml

```diff
@@ -16,19 +16,19 @@ jobs:
 
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
       fail-fast: false
 
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
-    - name: Install poetry
-      run: pipx install poetry
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
+    - name: Install poetry
+      run: pipx install --python ${{ matrix.python-version }} poetry
```

Comment:
Maybe use here pip with --break-system-packages as this is an isolated environment, to avoid issues with pipx and latest python version, as stated in the email.

Reply:
Thanks, @javihern98. Let's proceed as you suggested in your email, while I already merge this in the meantime.